### PR TITLE
feat(pagination): remove `pagination-icon.tsx` and use `@yamada-ui/lucide` icon

### LINKED ISSUE #3530: Replace `PaginationIcon` with `LucideIcon`
### Description

Replace `PaginationIcon` with `LucideIcon`.

### Problem Statement/Justification

Replace the icons from `@yamada-ui/icon` with those from `@yamada-ui/lucide`.

### Proposed Solution or API

_No response_

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

Yes

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by kaehehehe**
Closes #3530

## Description

remove `pagination-icon.tsx` and use `@yamada-ui/lucide` icon

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

File: packages/components/pagination/src/pagination-item.tsx [hunk 1]
```diff
@@ -60,11 +60,11 @@ export interface PaginationItemProps
 const iconMap: {
   [key in "ellipsis" | "first" | "last" | "next" | "prev" | number]: ReactNode
 } = {
-  ellipsis: <PaginationEllipsisIcon />,
-  first: <PaginationFirstIcon />,
-  last: <PaginationLastIcon />,
-  next: <PaginationNextIcon />,
-  prev: <PaginationPrevIcon />,
+  ellipsis: <EllipsisIcon viewBox="0 0 16 16" />,
```
Comment (hirotomoyamada) [inline packages/components/pagination/src/pagination-item.tsx:63]: Remove `viewBox`.

File: packages/components/pagination/src/pagination-item.tsx [hunk 2]
```diff
@@ -60,11 +60,11 @@ export interface PaginationItemProps
...
+  first: <ChevronsLeftIcon viewBox="0 0 16 16" />,
```
Comment (hirotomoyamada) [inline packages/components/pagination/src/pagination-item.tsx:64]: Remove `viewBox`.

File: packages/components/pagination/src/pagination-item.tsx [hunk 3]
```diff
@@ -60,11 +60,11 @@ export interface PaginationItemProps
...
+  last: <ChevronsRightIcon viewBox="0 0 16 16" />,
```
Comment (hirotomoyamada) [inline packages/components/pagination/src/pagination-item.tsx:65]: Remove `viewBox`.

File: packages/components/pagination/src/pagination-item.tsx [hunk 4]
```diff
@@ -60,11 +60,11 @@ export interface PaginationItemProps
...
+  next: <ChevronRightIcon viewBox="0 0 16 16" />,
+  prev: <ChevronLeftIcon viewBox="0 0 16 16" />,
```
Comment (hirotomoyamada) [inline packages/components/pagination/src/pagination-item.tsx:67]: Remove `viewBox`.

File: packages/components/pagination/src/pagination-item.tsx [hunk 5]
```diff
@@ -60,11 +60,11 @@ export interface PaginationItemProps
...
 } = {
-  ellipsis: <PaginationEllipsisIcon />,
-  first: <PaginationFirstIcon />,
-  last: <PaginationLastIcon />,
-  next: <PaginationNextIcon />,
-  prev: <PaginationPrevIcon />,
+  ellipsis: <EllipsisIcon viewBox="0 0 16 16" />,
+  first: <ChevronsLeftIcon viewBox="0 0 16 16" />,
+  last: <ChevronsRightIcon viewBox="0 0 16 16" />,
+  next: <ChevronRightIcon viewBox="0 0 16 16" />,
```
Comment (hirotomoyamada) [inline packages/components/pagination/src/pagination-item.tsx:66]: Remove `viewBox`.

Comment (kaehehehe) [reply packages/components/pagination/src/pagination-item.tsx]: Fixed. [89b1a34](https://github.com/yamada-ui/yamada-ui/pull/3531/commits/89b1a34abbbfc84e84b16702e49fb423e340fcda)

File: packages/components/pagination/src/pagination-item.tsx (hunk 4 above)
Comment (kaehehehe) [reply packages/components/pagination/src/pagination-item.tsx]: Fixed. [89b1a34](https://github.com/yamada-ui/yamada-ui/pull/3531/commits/89b1a34abbbfc84e84b16702e49fb423e340fcda)

File: packages/components/pagination/src/pagination-item.tsx (hunk 3 above)
Comment (kaehehehe) [reply packages/components/pagination/src/pagination-item.tsx]: Fixed. [89b1a34](https://github.com/yamada-ui/yamada-ui/pull/3531/commits/89b1a34abbbfc84e84b16702e49fb423e340fcda)

File: packages/components/pagination/src/pagination-item.tsx (hunk 2 above)
Comment (kaehehehe) [reply packages/components/pagination/src/pagination-item.tsx]: Fixed. [89b1a34](https://github.com/yamada-ui/yamada-ui/pull/3531/commits/89b1a34abbbfc84e84b16702e49fb423e340fcda)

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**